benchmark_stream_test.go

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"io"
+	"testing"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/yarpc/yab/encoding"
+	"github.com/yarpc/yab/protobuf"
+	"github.com/yarpc/yab/testdata/protobuf/simple"
+	"github.com/yarpc/yab/transport"
+)
+
+func TestBenchmarkMethodWarmTransportGRPCStreams(t *testing.T) {
+	tests := []struct {
+		name                                 string
+		procedure                            string
+		method                               string
+		numConnections                       int
+		requests                             [][]byte
+		returnOutput                         []simple.Foo
+		expectedInput                        []simple.Foo
+		expectedStreamsOpened                int32
+		expectedServerSentStreamMessages     int32
+		expectedServerReceivedStreamMessages int32
+	}{
+		{
+			name:                                 "client stream success",
+			procedure:                            "Bar::ClientStream",
+			method:                               "Bar/ClientStream",
+			numConnections:                       10,
+			requests:                             [][]byte{nil, nil},
+			returnOutput:                         []simple.Foo{{}},
+			expectedInput:                        []simple.Foo{{}, {}},
+			expectedStreamsOpened:                10,
+			expectedServerReceivedStreamMessages: 20,
+			expectedServerSentStreamMessages:     10,
```

Comment:
can you add some comments for the case? Took me some time to understand those abstract numbers :)
```
expectedServerReceivedStreamMessages: 20, // 10 client streams, each of them send 2 streams
expectedServerSentStreamMessages: 10, // but server responds a single message for each client stream
```